app/javascript/controllers/radio_behavior_controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from '@hotwired/stimulus'
+
+// Example:
+//
+// <div data-controller="radio-behavior" data-radio-behavior-selected-class="bg-red-400">
+//   <a href="#" data-action="radio-behavior#select" data-radio-behavior-target="radio">Link #1 acts like radio</a>
+//   <a href="#" data-action="radio-behavior#select" data-radio-behavior-target="radio">Link #2 acts like radio</a>
+// </div>
+//
+// Each time one of the radio elements is clicked, it will get the "selected" class and all the others will have
+// the class removed.
+
+
+export default class extends Controller {
+  static classes = [ "selected" ]
+  static targets = [ "radio" ]
+
+  select(event) {
+    this.radioTargets.forEach(element => {
+      element.classList.remove(...this.selectedClasses)
+    })
+
+    event.target.parentNode.classList.add(...this.selectedClasses)
```

Comment:
I'm surprised this needs to navigate up to the parent. Why is that? Does it need to match to make sure that the parentNode is one of the targets?

Reply:
It's good to point that out. That's brittle. It was intentional becuase I wand the radio class to be applied to a div which wraps the link href. I cannot put the data-action on the wrapping div because the href steals the click (I think because of stimulus).

I've updated the code to not just navigate to parent but search upward for the closest element. That will make this less brittle. And I updated the example to explain this.